libraries/pbes/include/mcrl2/pbes/detail/pbessolve_algorithm.h

```diff
@@ -251,6 +262,68 @@ class pbessolve_tool
   {
   }
 
+  void check_param_match(const mcrl2::data::variable_list Xparams, const mcrl2::data::variable_list X_hatparams)
+  {
+    if (Xparams.size() < X_hatparams.size())
+    {
+      throw mcrl2::runtime_error("An equation from the custom PBES has fewer parameters than the first PBES.");
+    }
+    for (mcrl2::data::variable p: X_hatparams)
+    {
+      auto it = std::find(Xparams.begin(), Xparams.end(), p);
+      if (it == Xparams.end())
+      {
+        throw mcrl2::runtime_error("An equation parameter from the first PBES was not found in the custom PBES.");
+      }
+    }
+  }
+
+  std::set<int> get_param_difference(const mcrl2::data::variable_list Xparams,
+    const mcrl2::data::variable_list X_hatparams)
+  {
+    std::set<int> R = {};
+    data::data_expression_vector params(Xparams.begin(), Xparams.end());
+    data::data_expression_vector params_hat(X_hatparams.begin(), X_hatparams.end());
+    check_param_match(Xparams, X_hatparams);
+
+    for (std::vector<mcrl2::data::data_expression>::size_type i = 0; i < params.size(); i++)
+    {
+      auto it = std::find(X_hatparams.begin(), X_hatparams.end(), params[i]);
+      if (it == X_hatparams.end())
+      {
+        mCRL2log(log::debug) << params[i] << " is redundant" << std::endl;
+        R.insert(i);
+      }
+    }
+    return R;
+  }
+
+  std::unordered_map<std::string, std::set<int>> construct_R(pbes_system::pbes first, pbes_system::pbes second)
```

Comment:
The other ones seem obvious, but perhaps some documentation on what the functions do might be appreciated.